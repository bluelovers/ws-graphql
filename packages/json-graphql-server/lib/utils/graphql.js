"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.printSchemaForTypes = exports.printSchemaForFields = void 0;
const graphql_1 = require("graphql");
/**
 * Return a schema string with a Main type using the fields
 *
 * @param {*array} fields
 *
 * @example
 * printSchemaForFields({
 *     id: { type: graphql.GraphQLString },
 *     title: { type: graphql.GraphQLString },
 *     views: { type: graphql.GraphQLInt },
 *     user_id: { type: graphql.GraphQLString },
 * });
 * // type Main {
 * //   id: String
 * //   title: String
 * //   views: String
 * //   user_id: String
 * // }
 * //
 * // type Query {
 * //   foo: Main
 * // }
 */
function printSchemaForFields(fields) {
    const mainType = new graphql_1.GraphQLObjectType({
        name: 'Main',
        fields,
    });
    const queryType = new graphql_1.GraphQLObjectType({
        name: 'Query',
        fields: {
            foo: { type: mainType },
        },
    });
    const schema = new graphql_1.GraphQLSchema({ query: queryType });
    return graphql_1.printSchema(schema);
}
exports.printSchemaForFields = printSchemaForFields;
function printSchemaForTypes(types) {
    const typesSchema = types.reduce((schema, type) => {
        schema[type.name] = type;
        return schema;
    }, {});
    const queryType = new graphql_1.GraphQLObjectType({
        name: 'Query',
        fields: types.reduce((fields, type) => {
            fields[type.name] = { type };
            return fields;
        }, {}),
    });
    const schema = new graphql_1.GraphQLSchema({ ...typesSchema, query: queryType });
    return graphql_1.printSchema(schema);
}
exports.printSchemaForTypes = printSchemaForTypes;
//# sourceMappingURL=graphql.js.map