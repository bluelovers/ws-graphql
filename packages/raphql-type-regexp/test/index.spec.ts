import { Kind, NameNode, ObjectValueNode } from 'graphql/language';
import GraphQLRegExpType, { serialize, parseValue, parseLiteral } from '../index';
import { ValueNode, StringValueNode } from 'graphql/language/ast';

describe('GraphQLRegExpType', () =>
{

	test(`serialize`, () =>
	{

		let actual;
		let expected = new RegExp(`g.*b`, 'i')

		actual = serialize(expected);
		expect(actual).toStrictEqual(expected.toString());
		expect(actual).toMatchSnapshot();

		actual = GraphQLRegExpType.serialize(expected);
		expect(actual).toStrictEqual(expected.toString());
		expect(actual).toMatchSnapshot();
	});

	test(`parseValue`, () =>
	{

		let actual;
		let expected = new RegExp(`g.*b`, 'i')

		actual = parseValue(expected.toString());

		expect(actual).toBeInstanceOf(RegExp);
		expect(actual).toEqual(expected);
		expect(actual).toMatchSnapshot();

		actual = GraphQLRegExpType.parseValue(expected.toString());

		expect(actual).toBeInstanceOf(RegExp);
		expect(actual).toEqual(expected);
		expect(actual).toMatchSnapshot();

	});

	test(`parseLiteral`, () =>
	{

		let actual;
		let expected = new RegExp(`g.*b`, 'i')

		let ast: ValueNode = {
			kind: Kind.STRING,
			value: '/g.*b/i',
		}

		actual = parseLiteral(ast);

		expect(actual).toBeInstanceOf(RegExp);
		expect(actual).toEqual(expected);
		expect(actual).toMatchSnapshot();

		actual = GraphQLRegExpType.parseLiteral(ast, null);

		expect(actual).toBeInstanceOf(RegExp);
		expect(actual).toEqual(expected);
		expect(actual).toMatchSnapshot();
	});

	test(`parseLiteral variables`, () =>
	{

		let actual;
		let expected = new RegExp(`g.*b`, 'i')

		let ast: ValueNode = {
			kind: Kind.VARIABLE,
			name: <NameNode>{
				value: 'regex',
			},
		}

		actual = parseLiteral(ast, {
			regex: '/g.*b/i',
		});

		expect(actual).toBeInstanceOf(RegExp);
		expect(actual).toEqual(expected);
		expect(actual).toMatchSnapshot();

		actual = GraphQLRegExpType.parseLiteral(ast, {
			regex: '/g.*b/i',
		});

		expect(actual).toBeInstanceOf(RegExp);
		expect(actual).toEqual(expected);
		expect(actual).toMatchSnapshot();
	});

	test(`parseLiteral object`, () =>
	{

		let actual;
		let expected = new RegExp(`g.*b`, 'i')

		let ast: ObjectValueNode = {
			kind: Kind.OBJECT,
			fields: [
				{
					kind: 'ObjectField',
					name: <NameNode>{
						value: 'source',
					},
					value: <StringValueNode>{
						kind: Kind.STRING,
						value: 'g.*b',
					},
				},
				{
					kind: 'ObjectField',
					name: <NameNode>{
						value: 'flags',
					},
					value: <StringValueNode>{
						kind: Kind.STRING,
						value: 'i',
					},
				},
			],
		}

		actual = parseLiteral(ast, null);

		expect(actual).toBeInstanceOf(RegExp);
		expect(actual).toEqual(expected);
		expect(actual).toMatchSnapshot();

		actual = GraphQLRegExpType.parseLiteral(ast, null);

		expect(actual).toBeInstanceOf(RegExp);
		expect(actual).toEqual(expected);
		expect(actual).toMatchSnapshot();
	});

})
