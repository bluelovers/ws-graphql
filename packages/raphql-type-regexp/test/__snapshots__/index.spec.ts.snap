// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQLRegExpType parseLiteral 1`] = `/g\\.\\*b/i`;

exports[`GraphQLRegExpType parseLiteral 2`] = `/g\\.\\*b/i`;

exports[`GraphQLRegExpType parseLiteral object 1`] = `/g\\.\\*b/i`;

exports[`GraphQLRegExpType parseLiteral object 2`] = `/g\\.\\*b/i`;

exports[`GraphQLRegExpType parseLiteral variables 1`] = `/g\\.\\*b/i`;

exports[`GraphQLRegExpType parseLiteral variables 2`] = `/g\\.\\*b/i`;

exports[`GraphQLRegExpType parseValue 1`] = `/g\\.\\*b/i`;

exports[`GraphQLRegExpType parseValue 2`] = `/g\\.\\*b/i`;

exports[`GraphQLRegExpType serialize 1`] = `"/g.*b/i"`;

exports[`GraphQLRegExpType serialize 2`] = `"/g.*b/i"`;
